### DISCUSSION TITLE
Local testing with MAMP PRO?

### DISCUSSION THREAD

**Discussion by yabdab** [spatie/ray#455] [Q&A]
local testing only randomly works when using MAMP Pro on mac Catalina. 

I will enable the server, then edit the settings, toggle it, etc.. then out of no where it starts working.

Most of the time it does not work no matter what I do.  ¯\_(ツ)_/¯ 

Is there something I am missing with this?

I am using the password method to connect.

server , port and username /password are correct.

**Comment by yabdab** [comment]
I guess this is not the place to get answers. 😕

**Reply by freekmurze** [reply]
We don't use Mamp Pro ourselves, so we can't give your specific support with that.

In essence the Ray app will listen for incoming messages on port 23517. The ray() function will send messages to that port.
Make sure that that port is free on your machine. If Mamp Pro run PHP in some kind of other environment (like docker) then make sure that you map the port in that environment to the local environment on your Mac.
